src/test/java/org/folio/rest/impl/CalculateOpeningTest.java

```diff
@@ -0,0 +1,295 @@
+
+    vertx.deployVerticle(RestVerticle.class, options, deploy -> {
+      try {
+        tenantClient.postTenant(null, post -> {
+          populateOpeningPeriods();
+          future.complete(null);
+        });
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    });
+
+    future.join();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    PostgresClient.stopEmbeddedPostgres();
+  }
+
+  private static void populateOpeningPeriods() {
+
+    OpeningHour openingHour = new OpeningHour();
+    openingHour.setStartTime("08:00");
+    openingHour.setEndTime("17:00");
+    List<OpeningHour> openingHours = Collections.singletonList(openingHour);
+
+    List<OpeningDayWeekDay> openingDays = new ArrayList<>();
+    openingDays.add(buildOpeningDayWeekDay(Weekdays.Day.MONDAY, openingHours));
+    openingDays.add(buildOpeningDayWeekDay(Weekdays.Day.TUESDAY, openingHours));
+    openingDays.add(buildOpeningDayWeekDay(Weekdays.Day.WEDNESDAY, openingHours));
+    openingDays.add(buildOpeningDayWeekDay(Weekdays.Day.THURSDAY, openingHours));
+    openingDays.add(buildOpeningDayWeekDay(Weekdays.Day.FRIDAY, openingHours));
+
+    //populate 3 periods for testing
+    //prev1     - January 2019, Monday-Friday, 8:00 - 17:00
+    //current1  - March   2019, Monday-Friday, 8:00 - 17:00
+    //next1     - May     2019, Monday-Friday, 8:00 - 17:00
+    createOpeningPeriod("prev1", LocalDate.of(2019, JANUARY, 1), LocalDate.of(2019, JANUARY, 31), openingDays);
+    createOpeningPeriod("current1", LocalDate.of(2019, MARCH, 1), LocalDate.of(2019, MARCH, 31), openingDays);
+    createOpeningPeriod("next1", LocalDate.of(2019, MAY, 1), LocalDate.of(2019, MAY, 31), openingDays);
+
+    //populate 3 more periods for testing with exceptional
+    //prev2     - January 2020, Monday-Friday, 8:00 - 17:00
+    //current2  - March   2020, Monday-Friday, 8:00 - 17:00
+    //next2     - May     2020, Monday-Friday, 8:00 - 17:00
+    createOpeningPeriod("prev2", LocalDate.of(2020, JANUARY, 1), LocalDate.of(2020, JANUARY, 31), openingDays);
+    createOpeningPeriod("current2", LocalDate.of(2020, MARCH, 1), LocalDate.of(2020, MARCH, 31), openingDays);
+    createOpeningPeriod("next2", LocalDate.of(2020, MAY, 1), LocalDate.of(2020, MAY, 31), openingDays);
+
+    //populate exceptional
+    createOpeningPeriod("prevExcptional", LocalDate.of(2020, FEBRUARY, 15), LocalDate.of(2020, FEBRUARY, 15),
+      Collections.singletonList(
+        buildOpeningDayWeekDay(mapLocalDateToString(
+          LocalDate.of(2020, FEBRUARY, 15)), true, true, false, null, openingHours)));
+
+    createOpeningPeriod("prevExcptional", LocalDate.of(2020, APRIL, 15), LocalDate.of(2020, APRIL, 15),
+      Collections.singletonList(
+        buildOpeningDayWeekDay(mapLocalDateToString(
+          LocalDate.of(2020, APRIL, 15)), true, true, false, null, openingHours)));
+  }
+
+  @Test
+  public void testCalculateOpenings() throws ParseException {
```

Comment:
There are 6 tests in this one test function. Could they be split up or have some kind of an output logged to the console to show what scenarios they are testing?

Reply:
Take a look at the method annotated with `@Parameterized.Parameters`. Each array in this method is one test case. We have 6 arrays and each element of the array is mapped to following variables `requestedDate`, `prevDate`, `currentDate`, `nextDate` in the order in which they are declared.

Reply:
I'd rather be explicit if possible, because a) when the tests failed it's hard to know exactly which one failed. I guess I could look at the order of the test and revisit this array and trace it out. b) newcomers to the code will have a hard time understand how the tests are constructed.

Reply:
Instead of a comment, make it a String parameter and log it at the start of the test. Would that make it clearer at run time?

Reply:
Thanks for the suggestion. Made corresponding changes.